### PR TITLE
New simulator block

### PR DESCRIPTION
**Pull request by VenusPasandi**
With this PR, we

1. add a new simulator block called `RobotDynWithContacts_closedChain` that is capable of
   - simulating a robot with multiple closed kinematic chains,
   - defining multiple links of the robot as the links interacting with the ground,
3. add a frictional impact model where we consider the friction effects in the impact model (Note that we still use the previous impact model by default. We can the frictional impact by setting the configuration parameter `useFrictionalImpact = true`),
4. add a discrete contact model where we consider the contact constraints in the velocity level. In the previous contact model, we consider the contact constraints in the acceleration level. (Note that we still use the previous contact model by default. We can use the discrete contact model by setting the configuration parameter `useDiscreteContact = true`),
5. add a criteria for the maximum acceptable consecutive fails in computing the reaction forces. If the solver fails more than the predefined criteria, the simulator block generates an error. Note that this criteria can be set through the configuration parameter `max_consecuitive_fail`.

Note that the PR does not change the high-level parameters of the `RobotDynWithContacts` block which released previously. So, the changes are aligned with the previous use of the repo.

### CODE REVIEW

**Review by traversaro — APPROVED** [review]
I just quickly went through the changes but the change seems contained and well documented.

**Review by Giulero — APPROVED** [review]

**Comment by nunoguedelha** [comment]
@VenusPasandi just submitted a few comments, just some typos.

File: README.md [hunk 1]
```diff
@@ -101,30 +101,51 @@ floatingBaseBalancingTorqueControlWithSimulator.torqueControlBalancingWithSimu
...
-|                                 |                     robotFrames  | [1×1] struct | Selected reference frames (base, left/right sole) |
-|<td rowspan="3">contact_config</td>|                     foot_print | [3×4] double | 4 Contact points on one of the feet soles |
+|                                 |                     robotFrames  | [1×1] struct  | Selected reference frames (base, left/right sole) |
+|                                 |                robotFrames.BASE  | char          | Base frame |
+|                                 |          robotFrames.LEFT_FOOT   | char         | (**RobotDynWithContacts** block) Left foot frame |
+|                                 |          robotFrames.RIGHT_FOOT  | char         | (**RobotDynWithContacts** block) Right foot frame |
+|                                 |robotFrames.IN_CONTACT_WITH_GROUND   | [1xN_C] cell | (**RobotDynWithContacts_closedChain** block) List of the links interacting with the ground |
+|                                 |          robotFrames.BREAK       | [1xN_P] struct  | (**RobotDynWithContacts_closedChain** block, optional) List of the prints of the break points in N_p closed chains |
```
[the remark this answers is not in the document]
Comment (VenusPasandi) [reply README.md:141]: Let us keep it as it is for now. I think by adding the proper documentation, the keywords will be clear.

File: README.md [hunk 2]
```diff
@@ -101,30 +101,51 @@ floatingBaseBalancingTorqueControlWithSimulator.torqueControlBalancingWithSimu
...
 |                                   |             total_num_vertices | [1x1] double | Total number of contact points (Nv=8) |
 |                                   |           friction_coefficient | [1x1] double | Ground/feet Coulomb friction coefficient $\mu _{xy}$ (F _{\bot} = \mu _{xy} N) |
+|                                   |           useFrictionalImpact  | boolean      | (optional) Consider the friction effects in the impact model if it is true. By default, it is false.     |
+|                                   |           useDiscreteContact   | boolean      | (optional) Use the discrete contact model that considers the contact constraints in the velocity level if it is true. By default, it is false.     |
+|                                   |           useQPOASES           | boolean      | (optional) Use QPOASES solver if it is true. By default, it is true. If it false, QUADPROG solver in MATLAB is used. |
+|                                   |          max_consecuitive_fail | boolean      | (optional) Maximum acceptable fails in computing the reaction forces. By default, it is equal to 10. |
```
[the remark this answers is not in the document]
Comment (VenusPasandi) [reply README.md]: it is resolved in d0f8de826bbc64b6687adf2db64dc23dd9c1b47c

**Review by nunoguedelha — APPROVED** [review]

**Pull request merged**